src/pudl/workspace/setup.py

```diff
@@ -4,310 +4,120 @@
-
-
-def init(pudl_settings: dict[str, str], clobber=False):
+    @property
+    def settings_dir(self) -> Path:
+        """Path to directory containing settings files."""
+        return self.input_dir.parent / "settings"
+
+    @property
+    def data_dir(self) -> Path:
+        """Path to PUDL data directory."""
+        # TODO(janrous): possibly deprecate this in favor of input_dir
+        return self.input_dir
+
+    @property
+    def pudl_db(self) -> Path:
+        """Returns url of locally stored pudl sqlite database."""
+        return self.sqlite_db("pudl")
+
+    def sqlite_db(self, name: str) -> str:
+        """Returns url of locally stored pudl slqlite database with given name.
+
+        The name is expected to be the name of the database without the .sqlite
+        suffix. E.g. pudl, ferc1 and so on.
+        """
+        db_path = PudlPaths().output_dir / f"{name}.sqlite"
+        return f"sqlite:///{db_path}"
+        return self.output_dir / f"{name}.sqlite"
+
+    def output_file(self, filename: str) -> Path:
+        """Path to file in PUDL output directory."""
+        return self.output_dir / filename
+
+    class Config:
+        """Pydantic configuration. Loads from .env file."""
+
+        env_file = ".env"
+        fields = {
+            "pudl_input": {"env": "PUDL_INPUT"},
+            "pudl_output": {"env": "PUDL_OUTPUT"},
+        }
+
+
+def init(clobber=False):
     """Set up a new PUDL working environment based on the user settings.
 
     Args:
-        pudl_settings (os.PathLike): Paths to data inputs & outputs. See
-            get_defaults() for how to get these.
         clobber (bool): if True, replace existing files. If False (the default)
             do not replace existing files.
 
     Returns:
         None
     """
     # Create tmp directory
-    tmp_dir = pathlib.Path(pudl_settings["data_dir"], "tmp")
+    tmp_dir = PudlPaths().data_dir / "tmp"
     tmp_dir.mkdir(parents=True, exist_ok=True)
 
     # These are files that may exist in the package_data directory, but that
     # we do not want to deploy into a user workspace:
     ignore_files = ["__init__.py", ".gitignore"]
 
+    # TODO(janrous): perhaps we don't need to do this?
     # Make a settings directory in the workspace, and deploy settings files:
-    settings_dir = pathlib.Path(pudl_settings["settings_dir"])
+    settings_dir = PudlPaths().settings_dir
     settings_dir.mkdir(parents=True, exist_ok=True)
     settings_pkg = "pudl.package_data.settings"
     deploy(settings_pkg, settings_dir, ignore_files, clobber=clobber)
 
     # Make output directory:
-    pudl_out = pathlib.Path(pudl_settings["pudl_out"])
-    pudl_out.mkdir(parents=True, exist_ok=True)
+    PudlPaths().output_dir.mkdir(parents=True, exist_ok=True)
+    # TODO(rousik): it might make sense to turn this into a method of
+    # PudlPaths object and to move this to settings.py from this module.
+    # Unclear whether deployment of settings files makes much sense.
```

Comment:
I personally would like to avoid deploying these settings files, but that definitely seems out of scope for this PR.

Reply:
Seems in-line with the effort of moving to data-only releases. Even with those, it's easy to point pudl at the settings files from the source repo IMO.